homeassistant/components/updater.py

```diff
@@ -0,0 +1,143 @@
+"""
+homeassistant.components.updater
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Indicates when a new update for HA is available
+"""
+# system imports
+import logging
+import os
+import requests
+import subprocess
+
+# homeassistant imports
+import homeassistant
+from homeassistant.helpers.entity import Entity
+
+# homeassistant constants
+DOMAIN = "updater"
+DEPENDENCIES = []
+REQUIREMENTS = ['requests>=2.0', 'gitpython>=1.0.1']
+CONF_PID_FILE = "pid_file"
+CONF_LOG_FILE = "log_file"
+CONF_REPO_NAME = "repo_name"
+CONF_REPO_BRANCH = "repo_branch"
+HA_SOURCE_DIR = os.path.abspath(os.path.join(homeassistant.__file__,
+                                             '..', '..'))
+GH_API_CALL = 'https://api.github.com/repos/{repo}/commits?sha={branch}'
+SCAN_INTERVAL = 60 * 60 * 12
+
+# setup logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """
+    Setup updater component.
+    """
+    # get optional configuration
+    pid_file = config['updater'].get(CONF_PID_FILE, None)
+    log_file = config['updater'].get(CONF_LOG_FILE, None)
+    repo_name = config['updater'].get(CONF_REPO_NAME, 'balloob/home-assistant')
+    repo_branch = config['updater'].get(CONF_REPO_BRANCH, 'master')
+
+    # create component entity
+    Updater(hass, _LOGGER, repo_name=repo_name, branch=repo_branch,
+            pid_file=pid_file, log_file=log_file)
+
+    return True
+
+
+class Updater(Entity):
+    """ Updater entity class """
+
+    hass = None
+    entity_id = 'updater.updater'
+    name = 'Updater'
+
+    def __init__(self, hass, logger, **kwargs):
```

Comment:
We always know how this class is being called, no need to add `**kwargs``. Only use the catch-alls when you need to